### PR TITLE
Minor fixes and corrections for ibtests

### PR DESCRIPTION
**Pull request by frankenmichl**
These commits refactor logging, and make sure all log data is uploaded in case the tests fail.

In addition, this contains minor fixes to the documentation to resemble current state of testsuite settings etc.

### CODE REVIEW

File: tests/kernel/ib_tests.pm [hunk 1]
```diff
@@ -42,19 +43,28 @@ sub check_dmesg {
...
 
     # wait until test is finished
     barrier_wait('IBTEST_DONE');
-
-    # just save the dmesg log
-    script_run("dmesg > /tmp/dmesg.txt");
-    upload_logs("/tmp/dmesg.txt");
-
-    # check dmesg for warnings etc.
-    check_dmesg;
```
[the remark this answers is not in the document]
Comment (frankenmichl) [reply tests/kernel/ib_tests.pm]: Hmm no. But, I should maybe remove the uploading of dmesg-output completely, as it is uploaded by default anyway. Just need to check the output for errors.

File: tests/kernel/ib_tests.pm [hunk 2]
```diff
@@ -42,19 +43,28 @@ sub check_dmesg {
...
+sub log_upload {
+    # just save the dmesg log
+    script_run("dmesg > /tmp/dmesg.txt");
+    upload_logs("/tmp/dmesg.txt");
+
+    # remove non-printable characters
+    script_run('tr -cd \'\11\12\15\40-\176\' < results/TEST-ib-test.xml > /tmp/results.xml');
+    parse_extra_log('XUnit', '/tmp/results.xml');
+
+    script_run("scp -o StrictHostKeyChecking=no root\@$slave:/tmp/dmesg.txt /tmp/dmesg_slave.txt");
```
Comment (jlausuch) [inline tests/kernel/ib_tests.pm]: I always like to use StrictHostKeyChecking=no  together with UserKnownHostsFile=/dev/null

File: tests/kernel/ib_tests.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/kernel/ib_tests.pm]: but does /tmp/dmesg.txt  exist on the slave? when do you create it?

**Comment by frankenmichl** [comment]
Need to change more, closing this PR

**Pull request closed without merging**